src/CsvHelper.Tests/CsvParserTests.cs

```diff
@@ -423,7 +423,41 @@ public void ParseRecordWhereOnlyCarriageReturnLineEndingIsUsed()
 			}
 		}
 
-		[TestMethod]
+	    [TestMethod]
+	    public void ParseRecordWhereOnlyCarriageReturnLineEndingIsUsedWithNextRowEmptyValue()
+	    {
+	        using (var memoryStream = new MemoryStream())
+	        using (var streamReader = new StreamReader(memoryStream))
+	        using (var streamWriter = new StreamWriter(memoryStream))
+	        using (var parser = new CsvParser(streamReader))
+	        {
+	            streamWriter.Write("one,two\r");
+	            streamWriter.Write(",four\r");
+	            streamWriter.Write("five,six\r");
+	            streamWriter.Flush();
+	            memoryStream.Position = 0;
+
+	            var record = parser.Read();
+	            Assert.IsNotNull(record);
+	            Assert.AreEqual(2, record.Length);
+	            Assert.AreEqual("one", record[0]);
+	            Assert.AreEqual("two", record[1]);
+
+	            record = parser.Read();
+	            Assert.IsNotNull(record);
+	            Assert.AreEqual(2, record.Length);
+	            Assert.AreEqual("", record[0]);
+	            Assert.AreEqual("four", record[0]);

```

Comment:
There's an error in your test, that should be `record[1]`. It works fine in master. I tested it because I just submitted a PR related to `\r` (#1523).